table/tables/tables.go

```diff
@@ -325,6 +325,10 @@ func (t *Table) AddRecord(ctx context.Context, r []types.Datum) (recordID int64,
 		if col.IsPKHandleColumn(t.meta) {
 			continue
 		}
+		if col.DefaultValue == nil && r[col.Offset].Kind() == types.KindNull {
```

Comment:
Check NotNil flag? If default value is not specified, col.DefaultValue will be nil.